nixos/modules/services/web-servers/traefik.nix

```diff
@@ -4,56 +4,111 @@ with lib;
 
 let
   cfg = config.services.traefik;
-  configFile =
-    if cfg.configFile == null then
+  jsonValue = with types; let
+    valueType = nullOr (oneOf [
+      bool
+      int
+      float
+      str
+      (lazyAttrsOf valueType)
+      (listOf valueType)
+    ]) // {
+      description = "JSON value";
+      emptyValue.value = {};
+    };
+  in valueType;
+  dynamicConfigFile =
+    if cfg.dynamicConfigFile == null then
       pkgs.runCommand "config.toml" {
         buildInputs = [ pkgs.remarshal ];
         preferLocalBuild = true;
       } ''
         remarshal -if json -of toml \
-          < ${pkgs.writeText "config.json" (builtins.toJSON cfg.configOptions)} \
+          < ${pkgs.writeText "dynamic_config.json" (builtins.toJSON cfg.dynamicConfigOptions)} \
+          > $out
+      ''
+    else cfg.dynamicConfigFile;
+  recursiveMerge = attrList:
```

Comment:
Is there a reason you couldn't use `lib.recursiveUpdate`? It seems to me it does the same thing, at least in this case.

Reply:
I wasn't aware of it, it looks indeed similar, thanks!